qtfred/src/ui/dialogs/ShipEditor/ShipWeaponsDialog.cpp

```diff
@@ -0,0 +1,181 @@
+#include "ShipWeaponsDialog.h"
+
+#include "ui_ShipWeaponsDialog.h"
+#include "WeaponsTBLViewer.h"
+
+#include <ui/util/SignalBlockers.h>
+#include <weapon/weapon.h>
+
+#include <QCloseEvent>
+#include <QStringListModel>
+namespace fso {
+namespace fred {
+namespace dialogs {
+ShipWeaponsDialog::ShipWeaponsDialog(QDialog* parent, EditorViewport* viewport, bool isMultiEdit)
+	: QDialog(parent), ui(new Ui::ShipWeaponsDialog()), _model(new ShipWeaponsDialogModel(this, viewport, isMultiEdit)),
+	  _viewport(viewport)
+{
+	ui->setupUi(this);
+
+	//Connect the mode change buttons
+	connect(ui->radioPrimary, &QRadioButton::toggled, this, [this](bool param) { modeChanged(param, 0); });
+	connect(ui->radioSecondary, &QRadioButton::toggled, this, [this](bool param) { modeChanged(param, 1); });
+	connect(ui->radioTertiary, &QRadioButton::toggled, this, [this](bool param) { modeChanged(param, 2); });
```

Comment:
One thing I think we should do in general is change all of these connect() lines to Qt slots. The pattern is usually something like `on_uiElementName_SIGNAL()`. So for these it'd be `on_radioPrimary_toggled(bool checked)`. The Qt docs can help figure out what the signals are and what their signatures are.

That said, I'll leave it up to you as to whether you want to do that refactor now as part of this PR or leave it to a follow-up. I'm good with either option.